src/extension/IOController_Windows.hpp

```diff
@@ -139,22 +185,37 @@ namespace extension {
                     }
                 }
 
-                // If not shudown
+                if (reactions_list_dirty || events_to_close.size() > 0) {
+                    // Get the lock so we don't concurrently modify the list
+                    std::lock_guard<std::mutex> lock(reaction_mutex);
+
+                    // Close any events we've queued for closing
+                    if (events_to_close.size() > 0) {
+                        for (auto& event : events_to_close) {
+                            if (!WSACloseEvent(event)) {
+                                throw std::system_error(
+                                    WSAGetLastError(), std::system_category(), "WSACloseEvent() failed");
+                            }
+                        }
 
-                // If dirty then update our list
+                        // Clear the queue of events to close
+                        events_to_close.resize(0);
```

Comment:
When you don't intend to reuse the memory use clear.
Clear will remove the backing memory, resize(0) leaves it untouched
```suggestion
                        events_to_close.clear();
```

Reply:
Nice, TIL!